presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClient.java

```diff
@@ -0,0 +1,89 @@
+
+import io.prestosql.plugin.jdbc.BaseJdbcClient;
+import io.prestosql.plugin.jdbc.BaseJdbcConfig;
+import io.prestosql.plugin.jdbc.ColumnMapping;
+import io.prestosql.plugin.jdbc.ConnectionFactory;
+import io.prestosql.plugin.jdbc.JdbcTypeHandle;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.type.VarcharType;
+
+import javax.inject.Inject;
+
+import java.math.RoundingMode;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.Optional;
+
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.decimalColumnMapping;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.varcharColumnMapping;
+import static io.prestosql.spi.type.DecimalType.createDecimalType;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.prestosql.spi.type.VarcharType.createVarcharType;
+import static java.lang.Math.max;
+
+public class OracleClient
+        extends BaseJdbcClient
+{
+    @Inject
+    public OracleClient(BaseJdbcConfig config, ConnectionFactory connectionFactory)
+    {
+        super(config, "\"", connectionFactory);
+    }
+
+    @Override
+    protected ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        return connection.getMetaData()
+                .getTables(null,
+                            schemaName.orElse(null),
+                            tableName.orElse(null),
+                            new String[] {"TABLE", "SYNONYM", "VIEW"});
+    }
+
+    @Override
+    public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        int columnSize = typeHandle.getColumnSize();
+        switch (typeHandle.getJdbcType()) {
+            case Types.CLOB:
+                return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+            case Types.NUMERIC:
+                int precision = columnSize == 0 ? 38 : columnSize;
+                int scale = max(typeHandle.getDecimalDigits(), 0);
+                return Optional.of(decimalColumnMapping(createDecimalType(precision, scale), RoundingMode.HALF_UP));
+            case Types.LONGVARCHAR:
+                if (columnSize > VarcharType.MAX_LENGTH || columnSize == 0) {
+                    return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+                }
+                return Optional.of(varcharColumnMapping(createVarcharType(columnSize)));
+        }
+        return super.toPrestoType(session, connection, typeHandle);
+    }
+
+    @Override
+    public PreparedStatement getPreparedStatement(Connection connection, String sql)
+            throws SQLException
+    {
+        PreparedStatement statement = connection.prepareStatement(sql);
+        statement.setFetchSize(1000);
```

Comment:
Why do you set that?

Reply:
this have to use fetchSize configuration
fixed